src/main/java/com/algorand/sdkutils/listeners/ResponseGenerator.java

```diff
@@ -38,7 +38,9 @@ public static void main(ResponseGeneratorArgs args, JCommander command) throws E
         }
 
         Publisher pub = new Publisher();
-        ResponseGenerator subscriber = new ResponseGenerator(args, pub);
+        ResponseGenerator subscriber = new ResponseGenerator(args);
+        pub.subscribeAll(subscriber);
```

Comment:
Why did you move the publisher out of the ResponseGenerator? Wouldn't the object be best suited to register the calls that it needs?

Reply:
The reason is the following: the constructor of an object should construct an object, and the actions on the object should be done outside the constructor. 

Calling a constructor and not doing anything with the object looks weird. It even triggers lint warning for creating an unused variable.